### DISCUSSION TITLE
Warning about preload not used because of script type mismatch

### DISCUSSION THREAD

**Discussion by allan-farrell** [ElMassimo/vite_ruby#45] [Q&A] [answered]
Is there any way to skip the preload tag generation? They just keep generating warnings in my app:

A preload for 'http://ratel.arts.mun.ca:3000/vite-dev/admin/application.js' is found, but is not used because the script type does not match.

Thanks for the gem!

A

**Comment by ElMassimo** [comment, marked as answer]
[fix]: https://github.com/rails/rails/pull/41697

Hi Allan!

Preload tags can be disabled by passing [`skip_preload_tags: true`](https://github.com/ElMassimo/vite_ruby/blob/main/vite_rails/lib/vite_rails/tag_helpers.rb#L24) to the [`vite_javascript_tag` helper](https://vite-ruby.netlify.app/guide/rails.html#tag-helpers-%F0%9F%8F%B7). However, it seems like you are running into a different scenario.

Rails generates [_early preload hints_](https://eileencodes.com/posts/http2-early-hints) incorrectly for `type="module"` scripts (uses `rel=preload` instead of [`rel=modulepreload`][fix]).

<img width="1438" alt="Screen Shot 2021-03-18 at 16 12 12" src="https://user-images.githubusercontent.com/1158253/111685545-ce76b700-8806-11eb-8732-ca0ea1a96570.png">

I made a [pull request to Rails fixing this][fix], and it has been merged, but I'm not sure [whether fixes are backported](https://twitter.com/MaximoMussini/status/1373644252207583238).

To disable that warning, you could:

- Upgrade to a version that includes [this patch][fix] (not released yet)
- Patch `javascript_include_tag` manually to include the [fix] (not ideal) 
- Disable early hints in development/production: `config.action_view.preload_links_header = false`
